browser/src/control/Control.JSDialogBuilder.js

```diff
@@ -2381,23 +2381,8 @@ L.Control.JSDialogBuilder = L.Control.extend({
 					}
 
 					if (disabled) {
-						if (data.command === '.uno:Paste') {
-							// Fix GitHub issue #5839 and never disable Paste toolbar button
```

Comment:
I had a concern that you do this unconditionally (even when this is not ios/android), but seems the fix was added for mobile and the scenario (start Writer, copy something on the clipboard, use paste button on the notetebookbar) does work even after this fix with a desktop browser, so probably we're OK.

Reply:
Back in 23.05, I assumed that the ".uno:Paste=enabled" message was never sent by LibreOffice on all platforms. I don't have a Linux machine so I had to make a guess and I guessed wrong. Fortunately, after digging into the LibreOffice code, it was clear that the iOS clipboard code in vcl/ios just had an incomplete implementation.

Might be worth checking if Android works too. I don't know if the Android native clipboard code detects changes in the system clipboard like Linux, macOS, and now iOS. But I think an easy way to test this is to do the following (either in Online or Android):
- Open a Writer document in editable mode (doesn't matter if the clipboard is empty or not)
- Switch to another app/application and copy something
- Back in the Writer document, press the Paste button in the Notebookbar and then select Paste Special in the popup menu
- If the pasted content matches what you copied from another app/application, then issue #5839 is not occurring

When issue #5839 occurred on iOS, either the Paste button would remain disabled (if the clipboard started empty) or the pasted content would be what was previously on the clipboard before copying from another app/application.

Reply:
Thanks, makes sense. I think Android/browsers don't have anything similar to the iOS PasteboardChangedEventListener / the usual desktop-style "owner changed" events, so what we do instead is to put a marker inside the HTML: if we get back our own HTML, we can do a much better internal paste, otherwise we do a HTML paste.